skbeam/core/tests/test_correlation.py

```diff
@@ -102,7 +115,20 @@ def test_lazy_multi_tau():
     for second_half_result in gen_second_half:
         pass
 
-    assert np.all(full_result.g2 == second_half_result.g2)
+    assert np.all(full_result.g2 ==
+                  second_half_result.g2)
+
+
+def test_two_time_corr():
+    setup()
+    y = []
+    for i in range(50):
+        y.append(img_stack[0])
+    two_time, lag_steps = two_time_corr(rois,
+                                        np.asarray(y), 50,
+                                        num_bufs=50, num_levels=1)
+    assert np.all(two_time)
```

Comment:
Added another test to take care of 1st half images correlated and then do the second half of images with the 1st half internal state and the compare with full set of images.